Convert containers hawkular Endpoints port=nil to port=443

cben opened the pull request:
For a time (before https://github.com/ManageIQ/manageiq-ui-classic/pull/1172), it was possible in UI to save containers provider with blank hawkular port, which got saved as port = nil.
Then it'd be impossible to edit the provider, as UI would crash:

![manageiq containers providers port nil](https://user-images.githubusercontent.com/273688/31744532-d33a9eda-b467-11e7-8a30-55d31f2b5d96.png)

On fine this was fixed by https://github.com/ManageIQ/manageiq-ui-classic/pull/1565 adding special case in UI to tolerate nil.
On master, @AparnaKarve [asked](https://github.com/ManageIQ/manageiq-ui-classic/pull/1305#issuecomment-309489658) to not introduce special cases but do a migration.
Better late than never :-)

After investigating, @yaacov @moolitayer and me concluded such provider does connect to hawkular with port 443 (it builds a url without port, confirmed experimentally).
This migration normalizes such endpoints to port 443, works same but possible to edit in UI:

![manageiq containers providers port 443](https://user-images.githubusercontent.com/273688/32183551-44961526-bda2-11e7-846d-0465257357cc.png)

Tested by checking out fine-1, creating such provider, migrating all the way to master, it was still broken, and fixed by this migration.  That's how I took above screenshots.
(in some cases such endpoints might have already been deleted by https://github.com/ManageIQ/manageiq/pull/14990 migration.  But not if hostname was autodetected from route => this is still sometimes needed.)

https://bugzilla.redhat.com/show_bug.cgi?id=1432070

@miq-bot add-label bug, data
@moolitayer @AparnaKarve please review

moolitayer commented:
> This migration normalizes such endpoints to port 0, still invalid but possible to edit in UI:

@cben what if we remove these endpoints altogether? that way we will not have failed authentication checks for an endpoint the user did not intend to create (event if the user does not go into the edit screen and disables metrics altogether)

cben commented:
These endpoints were only possible by actually going into Hawkular tab and
clearing the port field, not sure it's "didn't intend to create" but never
mind.
Yeah, I'm fine with deleting too.  Don't have an opinion, please decide for
me :)  cc @yaacov @zeari
Deleting will result in UI showing "Metrics: disabled" in Provider Edit
right?
In provider details, will it show as metrics error (ideal IMHO), or not
show metrics at all?

yaacov commented:
> it was possible in UI to save containers provider with blank hawkular port, which got saved as port = nil.

a. This is fine, when `port==nil` we fallback to default port [1] , IMHO, if you want to migrate it, it should be to the default port.

b. In old versions of ManageIQ it was not possible to have no metrics, metrics where always on, and values fall-back to default url. port and token if not given explicitly. IMHO migration of missing endpoint should be to a default values, and to nil/0.

[1]  it was possible in UI to save containers provider with blank hawkular port, which got saved as port = nil.

Fryguy commented:
This looks good to me, I just want approval from @simon3z or @moolitayer before merge.

simon3z commented:
> a. This is fine, when port==nil we fallback to default port [1] , IMHO, if you want to migrate it, it should be to the default port.

+1 for what @yaacov is suggesting here.
@cben what do you think?

> b. In old versions of ManageIQ it was not possible to have no metrics, metrics where always on, and values fall-back to default url. port and token if not given explicitly. IMHO migration of missing endpoint should be to a default values, and to nil/0.

This one is actually sad, but I don't have any other good suggestions here.

moolitayer commented:
> These endpoints were only possible by actually going into Hawkular tab and clearing the port field, not sure it's "didn't intend to create" but never mind.

I thought this was a situation where the user had pressed add from the `default` endpoint tab
I'm good with @yaacov's suggestion in this case :+1: 
I was really surprised that fine screens allowed that... (I checked that master blocks it so I can sleep at night)

> Deleting will result in UI showing "Metrics: disabled" in Provider Edit right?

Right

cben commented:
Confirmed experimentally, such endpoint is not broken as I originally thought but accesses https URL with port omitted, so it uses port 443.  This is also what @yaacov @moolitayer and me saw in code.

=> Changed migration to set port 443.  Dropped down migration, will not change 443 into nil.
Updated PR description and screenshot.  PTAL.

cben commented:
@yaacov PTAL

yaacov commented:
what about the update_all , don't you want to fix that first ?

o/w LGTM 👍

cben commented:
you mean miq-bot?  IIUC we routinely ignore this advice in migrations, because we want migrations to be fast where feasible, and because the model stubs don't normally have validations anyway.

@Fryguy please review (nice to have but not blocker)

Fryguy commented:
yeah, ignore the update_all warning (I keep forgetting to change that in the rubocop.yml)

@moolitayer You good with this?

chessbyte commented:
@simon3z @moolitayer can one of you guys approve this PR?
